### DISCUSSION TITLE
Developer-friendly starter theme

### DISCUSSION THREAD

**Discussion by adamchipperfield** [Shopify/dawn#3372] [Ideas]
Hello all!

Dawn is perfect for merchants, and sometimes good enough for simple developer customisation. But more than that and it's a royal pain.

**The documentation is sparse.** For example, there's nothing around how the grid system works. Don't get me started on the web components, which often are highly co-dependent. You often need to follow a looong chain to figure out what's going on.

**The code can be messy and inconsistent.** This one is probably more preference than anything. But it's crucial when developers want to extend the theme and are having to understand the way things are formatted. For example, some icons are inline whilst others are coming from a snippet.

**There's too many options.** I get it – Dawn is merchant-first and has to cover a massive spectrum of requirements. I don't have any issues with that, until I'm using Dawn for a client. It makes it excruciating to make changes since there's a 100 other side effects that will occur. That's 100 use cases I need to consider, when the client likely won't use any of them (but in the name of decency, I try to consider anyway).

**So what do I propose?** A "bare-bones" starter theme that puts the focus on extensibility (not to be confused with [Checkout Extensibility](https://help.shopify.com/en/manual/checkout-settings/checkout-extensibility)). The use case for this would be custom theme development projects. The [Skeleton Theme](https://github.com/Shopify/skeleton-theme) from the Slate days is probably the best example for this.

Here's the fundamentals:
- Cover all core templates and functionality (as little opinionated features as possible)
- Styling is basic (it shouldn't need to be stripped back, only added to)
- Documentation for everything
- Code is organised, tidy and consistent (use the "looks like it's been built by one dev" rule)
- _In the future:_ A library for common, but custom, features

I've been working with my own version of this for a good few years now, but it's hard to find the time to maintain it between projects. It would be ace if Shopify led something like this. And who knows, they might already be cooking something up.

Keep to hear feedback!

Adam

P.S. This is **not** a dig at the Dawn team or an op on its use in the non-custom space. They've built something that looks and functions great and has contributed to getting merchants selling quicker.

**Comment by PaulNewton** [comment]
Meta: dawn is **source-available**, vs open-source,  and that's gonna put the kibosh on a lot of things, so most the knowledge that's needed externally for the ecosystem to understand things with rigor stays undocumented, is in diaspora, or just plain internally silo'd and resisten to fundamental change even for the better.
               ----------------------------------

For sometime now the learning curve has had the ladder blown up from it so the amount of things new devs have to **just know**[1] continually gets steeper **and** high friction in comparison to a vintage theme like Debut that new devs may never even know existed.

Due to the complexity explosion also bear in mind even MORE upcoming changes to the theme architecture such as theme-blocks > block-hierarchy.
See the experimental reference theme
https://github.com/shopify/reference-theme
https://shopify.dev/docs/themes/architecture/blocks/theme-blocks

> A "bare-bones" starter theme that puts the focus on extensibility

Dawn is.. it's become alot.
Ideally needs to be 3 themes: an OS2.0 skeleton theme > a simple twilight(_pre-dawn_) theme based on the skeleton > and Dawn rebased ontop of the twilight theme[2]. And the themes need to aggressively refer to each other.

The twilight theme would introduce conventions for formatting and file organization and be the thing mentioned in the beginning of the architecture docs , i.e. [themes/architecture/templates/product](https://shopify.dev/docs/themes/architecture/templates/product).
Right now Dawn's [main-product.liquid](https://github.com/Shopify/dawn/blob/main/sections/main-product.liquid) is ALOT to parse through and is not something I like pointing newbie devs , **and especially not merchants**, towards as a "reference" implementation because it's pushing them into a growing deep end.
Dawn's grown into something that should be pointed to as an implementation _**standard**_ AFTER pedagogical introduction to something simpler and less of a moving target.

It needs to easier for new devs to literally step through building up a theme ( nothing > skeleton > twilight > snippets/sections/settings > dawn > production or theme-store), and for adopters to gauge how monolithic dawn is to facilitate paring down features and identify longtail issues such as with accessibility by just being able to diff twilight vs Dawn.


> Cover all [core](https://shopify.dev/docs/themes/architecture#templates:~:text=No%20templates%20are%20required.%20However%2C%20you%20need%20to%20have%20a%20matching%20template%20for%20any%20page%20type%20that%20you%20want%20to%20render.%20For%20example%2C%20to%20render%20a%20product%20page%2C%20you%20need%20at%20least%20one%20template%20of%20type%20product) templates and functionality (as little opinionated features as possible)

would need to define base sections (i.e. no sliders), and the reusable snippets of common features(i.e. price) , and bare settings functionality.
Settle on locales _if any_, whether to even include concepts like the section rendering api product recommendations, etc, and organizational conventions like making all the icons a singular or multiple file(s).

> Code is organized, tidy and consistent (use the "looks like it's been built by one dev" rule)

For the developer experience(DX): The WHY of **how** consistency is arrived at, needs to be defined better from the get go.
When your arm deep in a shopify theme your dealing with ~7 different naming conventions & formats between the 4 languages being used(HTML,CSS,JS,JSON,Liquid) and it' subsets(HTML>SVG). 
casing: _snake_case, kebab-case, lowerCamelCase, UpperCamelCase._ 
Idents: BEM 
_At least with dawn there's no framework like alpine,react, tailwind silliness, or weird CSS escaping ( .lg\:ml-0 ) also dumped into the mix for even more context switching._
_There is the liquid linter but that's a **post** effect information._

And like you noted there's non obvious conventions like why an SVG icon is inlined vs a snippet or even why all the icons are separate snippets, or why they aren't namespaced to show up at the end of the files list so we're not constantly scanning 40 files when looking for product snippets. I can understand why but the why's aren't defined for those without benefit of experience.


Still a limiting factor is no build systems , or asset include system, so CSS & JS still tends to be monolithic & idiomatic (see .rte etc) for shopify behaviors.

[1] Did you know the dawn theme uses BEM for class names, do you even know what BEM is, is it strict in the theme? do you even know a single place that documents that Dawn uses BEM?
[2] twilight is just dawnv1 with relevant bug fixes?? 🤷

**Reply by PaulNewton** [reply]
Update,  Shopify's released an updated version of their first party Skeleton theme https://github.com/Shopify/skeleton-theme
https://shopify.dev/changelog/skeleton-theme-is-now-available
skeleton theme is in the cli via `shopify theme init`
Other features making themes more developer friendly: section labeling, copy&paste in the theme editor , and shopify-magic can generate custom blocks,  etc
_As part of [shopify editions summer 2025 #Developer](https://www.shopify.com/editions/summer2025#developer) , and see the dev changelog https://shopify.dev/changelog?filter=dev_themes_

## Blah blah "developer friendly"
Developer friendly == you already know what your doing in shopify theme land but don't have time to build every single little tool.

The skeleton theme is more of a boilerplate theme. Almost be more preferable to have a first-party script||app that can progressively gut existing themes down to bones instead of everyone making one-offs; _🤔 but stripping down systems manually can help new learners ALOT 🤷‍♂️_.
 So I don't know if I would call such a thing "developer friendly" without the implication that the "developer" already needs to KNOW HOW shopify theme architecture works , liquid quirks, etc and not need ANY guidance to build on top of it.
 With each years bells and whistles pulling the ladder up further in a not so friendly fashion hastened by a lack of graduated levels in theme docs(e.g. make a 1.0 'simple' theme, make an ['advanced' 2.0 theme](https://github.com/Shopify/dawn)) .
An unfriendly side-effect of JSON based systems with strict schema is ending up with a **COMPLETE LACK** of guiding commentary for new learners, so it has lots of context switching built in <ins>as a feature</ins>;  _🤓even if there are comments lots of systems in shopify land LOVE to minify JSON without recourse_.

📋🧠The new skeleton theme is missing things like gift-card layout, and it doesn't provide any /templates/*.liquid templates to contrast vs .JSON templates, no static section usage aside from gorups, and doesn't use a convention of main-<resource-or-context-name> for primary sections which can cause some learning friction if comparing to "real" themes.
New learners won't even know such concepts exist, not developer friendly when viewed that way.
Arguable if those missing things are opportunities for learning, or repo-contributions or not.


Tangentially , for some use cases, there is new web-components to help rapidly iterate in a "developer friendly" fashion, though generally such code is intend for other platforms, ala buy-buttons sdk , and not shopify's online-sales channel.
- docs https://shopify.dev/custom-storefronts/tools/web-components
- playground https://webcomponents.shopify.dev/playground?view=editor
- **_bloated_** ~landing~ marketing page https://webcomponents.shopify.dev
🤔platforms doing LLMs.text is the saddest most interesting part https://webcomponents.shopify.dev/llms.txt , its an absurdity it's taken the market-forces of machine-learning to get shopify to provide simple **clean** .txt docs for an api that is ironically not even meant for humans but super usable by humans in-place of the heavy bloated glossy docs meant for humans where aesthetics was made more important than utility , actual use cases, ad nauseum🤪

**Comment by tnoetzel** [comment]
It's absolutely crazy to me that this doesn't already exist. Since it seems like Shopify isn't prioritizing this, if anybody is interested in collaborating on _creating it_ I'd definitely be down.

**Comment by pablogiralt** [comment]
I am in on collaborating on this too!

**Comment by ketanmistry** [comment]
I've spent a lot of time creating a boilerplate theme because I find Dawn quite hard to work with when clients have asked for some special features to be added. 

I've never built a site using a pre-made theme as I find it takes me more time, but I often get asked to add features to stores that I've not built - all of them have always been pre-made themes. I never enjoy working on them.

If you're interested you can find out more about it at https://taillourtheme.com. It's built with some additional features that clients have asked me for over the last decade.

**Reply by elliotkidd** [reply]
would be keen to check this out but looks like your site is down?

**Reply by ketanmistry** [reply]
Hi @elliotkidd, it's back up and running - I was moving it over to GitHub Pages. 🙂

**Reply by aanomm** [reply]
amazing work @ketanmistry 🤩. thanks for making it available 🙇🏾

**Reply by avcohen** [reply]
This is great. Thanks for your contribution!
